core/new-gui/src/app/workspace/component/navigation/navigation.component.ts

```diff
@@ -261,35 +273,38 @@ export class NavigationComponent implements OnInit {
   }
 
   /**
-   * Returns true if there's any operator on the graph; false otherwise
+   * Persist the cachedWorkflow in the localStorage, save it to the backend database. It also fetches the new workflow.wid
+   * and updates it in the cache if this is a new workflow.
    */
-  public hasOperators(): boolean {
-    return this.workflowActionService.getTexeraGraph().getAllOperators().length > 0;
-  }
-
-  public onClickSaveWorkflow(): void {
-    if (!this.userService.isLogin()) {
-      alert('please login');
-    } else {
-      const cachedWorkflow: Workflow | null = this.cachedWorkflowService.getCachedWorkflow();
-      if (cachedWorkflow != null) {
-        this.isSaving = true;
-        this.workflowPersistService.persistWorkflow(cachedWorkflow).subscribe(this.cachedWorkflowService.cacheWorkflow).add(() => {
-          this.isSaving = false;
-        });
-      } else {
-        alert('No workflow found in cache.');
-      }
+  public persistCachedWorkflow(): void {
+    const cachedWorkflow: Workflow | undefined = this.workflowCacheService.getCachedWorkflow();
+    if (cachedWorkflow !== undefined) {
+      this.isSaving = true;
+      this.workflowPersistService.persistWorkflow(cachedWorkflow).subscribe((updatedWorkflow: Workflow) => {
```

Comment:
is it better to let `workflowPersistService` fetch the current workflow by itself instead of taking it as a parameter?

Reply:
I thought of this. But I want those two services to be independent of each other.

Reply:
This is back to the "single source of truth" thing, if the workflow cache service is your single source of truth, then everyone should ask the "truth" from it